### PR TITLE
This is a commit for Assert Login Page Assert, Add, Edit and Delete License

### PR DESCRIPTION
**Pull request by prpatel1011**

### CODE REVIEW

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    private final String LicDeleteYesBtn = "(//*[@class='oxd-button oxd-button--medium oxd-button--label-danger orangehrm-button-margin'][@type='button'])";
+
+    private final String getIdLicCancelBtn = "//*[@id=\"app\"]/div[1]/div[2]/div[2]/div/div/form/div[2]/button[1]";
+
+    public LicensesPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public boolean SaveNewLicenses(String LicName) {
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: method name should start with small letter

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        new WebDriverWait(driver, Duration.ofSeconds(5))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.xpath(Banner)));
+        boolean dis = driver.findElement(By.xpath(Banner)).isDisplayed();
+        System.out.println("Green banner with Success message displayed: " + dis);
+        return dis;
+
+
+    }
+
+    public boolean EditLicense(String LicName1) {
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: method name should start with small letter

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        driver.findElement(By.xpath(LicEditNameBtn)).sendKeys(LicName1);
+        click(By.xpath(LicEditSaveBtn));
+        boolean dis = driver.findElement(By.xpath(Banner)).isDisplayed();
+        System.out.println("Green banner with Success message displayed: " + dis);
+        return dis;
+
+    }
+
+
+    public boolean DeleteLicense() {
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: method name should start with small letter

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class LicensesPage extends PageBase {
+
+    private final String LicAddBtn = "//*[@id=\"app\"]/div[1]/div[2]/div[2]/div/div/div[1]/div/button";
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: make the xpath shorter and understandable like the rest

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    private final String LicTextName = "//*[@id=\"app\"]/div[1]/div[2]/div[2]/div/div/form/div[1]/div/div[2]/input";
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: change xpath

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    private final String LicSaveBtn = "//button[@type='submit']";
+    private final String Banner = "//*[@id=\"oxd-toaster_1\"]/div";
+
+    private final String LicEditSaveBtn = "//button[@type='submit']";
+
+    private final String LicPenBtn = "(//*[@class='oxd-icon-button oxd-table-cell-action-space'][@type='button'])[10]";
+
+    private final String LicEditNameBtn = "//*[@id=\"app\"]/div[1]/div[2]/div[2]/div/div/form/div[1]/div/div[2]/input";
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: change xpath

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    private final String LicDeleteBtn = "(//*[@class='oxd-icon-button oxd-table-cell-action-space'][@type='button'])[9]";
+
+    private final String LicDeleteYesBtn = "(//*[@class='oxd-button oxd-button--medium oxd-button--label-danger orangehrm-button-margin'][@type='button'])";
+
+    private final String getIdLicCancelBtn = "//*[@id=\"app\"]/div[1]/div[2]/div[2]/div/div/form/div[2]/button[1]";
```
Comment (jemsyjose) [inline src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: change xpath

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 1 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 2 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 3 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 4 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 5 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 6 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

File: src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java (hunk 7 above)
Comment (prpatel1011) [reply src/test/java/ca/qaguru/oranghrmbatch24/pages/LicensesPage.java]: Complied comment

**Review by jemsyjose — APPROVED** [review]
Approved Changes

**Pull request merged**